### PR TITLE
Use Swift's native `String` instead of `NSString` based `String`

### PR DESCRIPTION
**Pull request by norio-nomura**
- Add `sourceKittenRawValueStringFrom(_:)` and comments for #148
- By applying this,The duration of linting Carthage 0.11 by SwiftLint(8af5c4e)
 with SourceKitten(6025179) is reduced from:
```
swiftlint lint --config ~/.swiftlint-test.yml  18.34s user 1.14s system 83% cpu 23.271 total
```
to:
```
swiftlint lint --config ~/.swiftlint-test.yml  17.60s user 1.13s system 85% cpu 22.014 total
```

Adding enums mentioned in #148 will be separated PR.

### CODE REVIEW

File: Source/SourceKittenFramework/Request.swift [hunk 1]
```diff
@@ -103,16 +104,59 @@ Cache SourceKit requests for strings from UIDs
...
+ 
+- parameter bytes: UnsafePointer<Int8>
+- parameter length: length of bytes
+
+- returns: String Swift's native String
+*/
+private func swiftStringFrom(bytes: UnsafePointer<Int8>, length: Int) -> String? {
+    let pointer = UnsafeMutablePointer<Int8>(bytes)
+    return String(bytesNoCopy: pointer, length: length, encoding: NSUTF8StringEncoding,
+        freeWhenDone: false).map { "\($0)" }
```
Comment (jpsim) [inline Source/SourceKittenFramework/Request.swift]: Why the `map`? It just goes from `String?` to `String?`.

Comment (neilpa) [inline Source/SourceKittenFramework/Request.swift]: It's converting from the `NSString` backed swift string to a native swift string. The c'tor is actually `-[NSString initWithBytesNoCopy:length:encoding:freeWhenDone]`

Comment (jpsim) [inline Source/SourceKittenFramework/Request.swift]: Awesome.

**Comment by jpsim** [comment]
Thanks for this!

**Comment by norio-nomura** [comment]
:pray:

**Pull request merged**